src/scripts/activate-user.ts

```diff
@@ -90,11 +89,67 @@ const priorToActivationUserStatusCheck = (
     }
     // STAGED or DEPROVISIONED user status
     default: {
-      return TE.right(user);
+      return TE.right({
+        ...user,
+        status: user.status,
+      });
     }
   }
 };
 
+/**
+ * Prints out what would happen if the user was activated.
+ * @param eitherUser - either the user or an error message.
+ * @returns a TaskEither that resolves to the user.
+ */
+const dryRunActivateUser = (
+  eitherUser: TE.TaskEither<string, ActivatableUser>
+): TE.TaskEither<string, User> =>
+  pipe(
+    eitherUser,
+    TE.tapIO((user) =>
+      Console.info(`Will attempt to activate [${user.id}] [${user.email}].`)
+    )
+  );
+
+/**
+ * Activates the user.
+ * @param service - the service to use to activate the user.
+ * @param eitherUser - either the user or an error message.
+ * @returns a TaskEither that resolves to the activated user.
+ */
+const activateUser = (
+  service: UserService,
+  eitherUser: TE.TaskEither<string, ActivatableUser>
```

Comment:
**Issue: ** this shouldn't be a `TE`. It should be an `ActivatableUser`. In *general* you want the function 'higher' in the call stack to managing the 'failure' side.

Reply:
fixed in commit: 24c1ffb0115047966275f9de038f03259c636f52